lib/memory.h

```diff
@@ -18,6 +18,15 @@ extern "C" {
 #define SF_REALLOC(p, s) sf_realloc(p, s, __FILE__, __LINE__)
 #define SF_FREE(p) ((void) (sf_free(p, __FILE__, __LINE__), (p) = NULL))
 
+typedef struct SF_INTERNAL_MEM_HOOKS {
+    void *(*alloc)(size_t size);
+    void (*dealloc)(void *ptr);
+    void *(*realloc)(void *ptr, size_t size);
+    void *(*calloc)(size_t nitems, size_t size);
+} SF_INTERNAL_MEM_HOOKS;
+
+static SF_INTERNAL_MEM_HOOKS global_hooks = {malloc, free, realloc};
```

Comment:
calloc?

Reply:
@notkriswagner calloc is missing.

Reply:
Yup, I added calloc into the struct afterward, forgot to add it to the initialization.